Tests/SentryTests/SentryCrash/SentryCrashWrapperTests.swift

```diff
@@ -0,0 +1,125 @@
+@_spi(Private) @testable import Sentry
+import XCTest
+
+final class SentryCrashWrapperTests: XCTestCase {
+    
+    private var crashWrapper: SentryCrashWrapper!
+    private var scope: Scope!
+    
+    override func setUp() {
+        super.setUp()
+        crashWrapper = SentryCrashWrapper(processInfoWrapper: ProcessInfo.processInfo,
+            systemInfo: [
+            "osVersion": "23A344",
+            "kernelVersion": "23.0.0",
+            "isJailbroken": false,
+            "systemName": "iOS",
+            "cpuArchitecture": "arm64",
+            "machine": "iPhone14,2",
+            "model": "iPhone 13 Pro",
+            "freeMemorySize": UInt64(1_073_741_824), // 1 GB
+            "usableMemorySize": UInt64(4_294_967_296), // 4 GB
+            "memorySize": UInt64(6_442_450_944), // 6 GB
+            "appStartTime": "2023-01-01T12:00:00Z",
+            "deviceAppHash": "abc123",
+            "appID": "12345",
+            "buildType": "debug",
+            "CFBundleIdentifier": "io.sentry.crashTest",
+            "CFBundleName": "CrashSentry",
+            "CFBundleVersion": "201702072010",
+            "CFBundleShortVersionString": "1.4.1"
+        ])
+        scope = Scope()
+    }
+    
+    override func tearDown() {
+        crashWrapper = nil
+        scope = nil
+        super.tearDown()
+    }
+    
+    func testEnrichScope_WithSystemInfo_SetsOSContext() {
+        crashWrapper.enrichScope(scope)
+        
+        let osContext = scope.contextDictionary["os"] as? [String: Any]
+        XCTAssertNotNil(osContext)
+        XCTAssertNotNil(osContext?["name"])
+        XCTAssertNotNil(osContext?["version"])
```

Comment:
`h`: This returns an empty string, because the `SentryUIDeviceWrapper.getSystemVersion()` returns an empty string. It would be great to do proper assertions here and not simply assert it's not `nil`. Either the test setup is wrong or we have a bug.

Reply:
When running this tests only , it seems like it was empty, fixing that